File: instrumentation/play-ws-2.6.0/src/main/scala/com/nr/agent/instrumentation/playws/InboundWrapper.scala
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.playws

import com.newrelic.api.agent.{ExtendedInboundHeaders, HeaderType}
import play.api.libs.ws.StandaloneWSResponse

import scala.collection.JavaConversions

class InboundWrapper(var response: StandaloneWSResponse) extends ExtendedInboundHeaders {

  override def getHeaderType = {
    HeaderType.HTTP
  }

  override def getHeader(name: String) = {
    response.header(name).orNull
  }

  override def getHeaders(name: String): java.util.List[String] = {
    val result = response.headerValues(name)
    if (result.isEmpty) {
      return null
    }
    JavaConversions.seqAsJavaList(result)
  }

}

File: instrumentation/okhttp-4.4.0/src/main/java/com/nr/agent/instrumentation/okhttp44/InboundWrapper.java
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.okhttp44;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.ExtendedInboundHeaders;
import okhttp3.Headers;
import okhttp3.Response;

import java.util.List;

public class InboundWrapper extends ExtendedInboundHeaders {

    private final Headers headers;

    public InboundWrapper(Response response) {
        this.headers = response == null ? null : response.headers();
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public String getHeader(String name) {
        return headers.get(name);
    }

    @Override
    public List<String> getHeaders(String name) {
        return headers.values(name);
    }

}
